packages/core/src/storage.ts

```diff
@@ -1,50 +1,36 @@
 import type { Readable } from "stream";
 import { isReadableStream } from "./utils";
+import {
+  StorageRequestStatus,
+  getSdk,
+  StatusByIdQuery,
+  StatusByCidQuery,
+} from "./client";
+import { GraphQLClient } from "graphql-request";
 
-/**
- * Status is the status of a StorageRequest.
- * * `Unknown` is the default value to an unitialized StorageRequest. This status must be
- *   considered invalid in any real StorageRequest instance.
- * * `Batching` indicates that the storage request is being batched.
- * * `Preparing` indicates that the batch containing the data is being prepared.
- * * `Auctioning` indicates that the batch containing the data is being auctioned.
- * * `DealMaking` indicates that the data is in deal-making process.
- * * `Success` indicates that the request was stored in Filecoin.
- * * `Error indicates that there is some error handling the request.
- */
-type Status =
-  | "Unknown"
-  | "Batching"
-  | "Preparing"
-  | "Auctioning"
-  | "DealMaking"
-  | "Success"
-  | "Error";
-
-/**
- * Request is a request for storing data in a Provider.
- */
-export interface Request {
+export interface StorageRequest {
   id: string;
   cid: {
     "/": string;
   };
-  status_code: Status;
+  status: StorageRequestStatus;
 }
 
+type RequestAlias = Omit<StorageRequest, "status"> & { status_code: string };
```

Comment:
right now, upload is out of sync with the GraphQL api enum values, so that's fine, we deal with that below